src/Features/CSharp/Portable/BraceCompletion/AbstractCurlyBraceOrBracketCompletionService.cs

```diff
@@ -115,18 +114,22 @@ private static bool ContainsOnlyWhitespace(SourceText text, int openingPosition,
             if (closingPointLine - openingPointLine == 1)
             {
                 // Handling syntax tree directly to avoid parsing in potentially UI blocking code-path
-                var closingToken = document.Root.FindTokenOnLeftOfPosition(context.ClosingPoint);
-                var newLineString = options.FormattingOptions.NewLine;
-                newLineEdit = new TextChange(new TextSpan(closingToken.FullSpan.Start, 0), newLineString);
+                var closingToken = document.Root.FindToken(closingPoint - 1);
 
-                var generator = document.LanguageServices.GetRequiredService<SyntaxGeneratorInternal>();
-                var endOfLine = generator.EndOfLine(newLineString);
+                var newClosingToken = closingToken.WithPrependedLeadingTrivia(
+                    SpecializedCollections.SingletonEnumerable(SyntaxFactory.EndOfLine(options.FormattingOptions.NewLine)));
 
-                var rootToFormat = document.Root.ReplaceToken(closingToken, closingToken.WithPrependedLeadingTrivia(endOfLine));
-                document = document.WithChangedRoot(rootToFormat, cancellationToken);
+                var rootToFormat = document.Root.ReplaceToken(closingToken, newClosingToken);
+
+                newClosingToken = rootToFormat.FindToken(closingPoint - 1);
```

Comment:
instead of this, use an annotation.  that will be *much* safer and more correct.  i really just do not trust thsi code.

Reply:
Sure. Fixed